modules/openapi-generator/src/main/resources/cpp-rest-sdk-client/cmake-lists.mustache

```diff
@@ -27,28 +25,55 @@ endif()
 # BUILD TYPE
 message("A ${CMAKE_BUILD_TYPE} build configuration is detected")
 
-# Update require components as necessary
-#find_package(Boost 1.45.0 REQUIRED COMPONENTS ${Boost_THREAD_LIBRARY} ${Boost_SYSTEM_LIBRARY} ${Boost_REGEX_LIBRARY} ${Boost_DATE_TIME_LIBRARY} ${Boost_PROGRAM_OPTIONS_LIBRARY} ${Boost_FILESYSTEM_LIBRARY})
+if (UNIX)
+
+message(STATUS "Building client library for Linux/Unix")
+
+set(CMAKE_BUILD_TYPE Debug)
+set(CMAKE_C_FLAGS   "${CMAKE_C_FLAGS}   -fPIC -Wall -Wno-unused-variable")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC -Wall -Wno-unused-variable")
+
+find_package(cpprestsdk REQUIRED)
+find_package(Boost REQUIRED)
 
-# build and set path to cpp rest sdk
-set(CPPREST_ROOT ${PROJECT_SOURCE_DIR}/3rdParty/cpprest)
-set(CPPREST_INCLUDE_DIR ${CPPREST_ROOT}/include)
-set(CPPREST_LIBRARY_DIR ${CPPREST_ROOT}/lib)
+else(UNIX)
 
-include_directories(${PROJECT_SOURCE_DIR} api model ${CPPREST_INCLUDE_DIR})
+message(STATUS "Building client library for Windows")
 
+find_package(cpprestsdk REQUIRED)
+find_package(Boost REQUIRED)
+find_package(pthreads REQUIRED)
 
-# If using vcpkg, set include directories. Also comment out CPPREST section above since vcpkg will handle it.
-# To install required vcpkg packages execute:
-# > vcpkg install cpprestsdk cpprestsdk:x64-windows boost-uuid boost-uuid:x64-windows
-# set(VCPKG_ROOT "C:\\vcpkg\\installed\\x64-windows")
-# set(VCPKG_INCLUDE_DIR ${VCPKG_ROOT}/include)
-# set(VCPKG_LIBRARY_DIR ${VCPKG_ROOT}/lib)
-# include_directories(${PROJECT_SOURCE_DIR} api model ${VCPKG_INCLUDE_DIR})
+endif(UNIX)
+
+# Manually set the cpprestsdk paths
+#set(CPPREST_INCLUDE_DIR ${CPPREST_ROOT}/include)
+#set(CPPREST_LIBRARY_DIR ${CPPREST_ROOT}/lib)
+#include_directories(${CPPREST_INCLUDE_DIR})
+#link_directories(${CPPREST_LIBRARY_DIR})
+
+include_directories(
```

Comment:
I would use more modern target_include_directories - this would lead to less clutter if someone decides to do `add_subdirectory` on a directory containing this file - it would propagate and everything declared after would have these directories.

If projects linking to `${PROJECT_NAME}` require to know these headers then do `target_include_directories(${PROJECT_NAME} PUBLIC ...)`, if not then change `PUBLIC` with `PRIVATE`